src/test/java/com/puppycrawl/tools/checkstyle/filters/AbstractPatchFilterEvaluationTest.java

```diff
@@ -91,7 +92,31 @@ protected void testByConfig(String configPath)
             final String expectedFile = configPath.replaceFirst(
                     "(default|zero)ContextConfig.xml", "expected.txt");
             final List<String> expected = Files.readAllLines(Paths.get(getPath(expectedFile)));
-            final List<String> actuals = lnr.lines().limit(expected.size()).sorted()
+            final Comparator<String> comparator = new Comparator<String>() {
```

Comment:
this solution is overcomplicated and forces to keep certain order in expected.txt (without telling what is wrong), especially non-obvious one like Test2.java should be before Test.java. Order of expected messages should not matter.
Basicaly you need to check:
 - amounts of expected and actual errors are the same
 - sets (not lists) of expected and actual errors are the same

Reply:
> sets (not lists) of expected and actual errors are the same

also instead of sets comparing, you can order `expected` list:
```java
expected.sort(Comparator.naturalOrder());
```

Reply:
> also instead of sets comparing, you can order expected list (because actuals is sorted):

done

> amounts of expected and actual errors are the same

I think this one have checked amounts.
```
assertEquals("unexpected output: " + lnr.readLine(),
                    expected.size(), errorCounter);
```

Reply:
ok, this should work